plugins/destination/mysql/client/schema.go

```diff
@@ -13,20 +13,13 @@ func identifier(name string) string {
 	return fmt.Sprintf("`%s`", name)
 }
 
-const maxPrefixLength = 191
-
 const columnQuery = `SELECT 
 cols.COLUMN_NAME,
 COLUMN_TYPE,
 IS_NULLABLE,
-constraint_type,
-sub_part
```

Comment:
I couldn't understand why we need `sub_part`. Please comment if you know why

Reply:
https://github.com/cloudquery/cloudquery/pull/11982#discussion_r1255691867

Reply:
Saw that comment and still don't understand the reason. It is because we want the migration to fail?

Reply:
I presume so.
I think @bbernays is the one who can answer this question precisely.

Reply:
Yes, without it migrations wouldn't catch (and then update) the subpart for existing DBs. Not sure if this was the best tradeoff (as I agree that it inhibits the DB->DB) use case... 

But by shortening the subpart, we are risking losing data in cases when the beginning of PKs are the same...

Reply:
> But by shortening the subpart, we are risking losing data in cases when the beginning of PKs are the same

Yeah, but we won't be able to create the table otherwise as we would exceed the allowed limit to the sync won't work at all. At least with the shorter prefix we'll have the log warning about truncation

Reply:
Also, please note that with this fix, a single PK will have a longer prefix than `191`

Reply:
Those won't really help source DB -> MySQL users

Reply:
It will help all but mysql-mysql right? Which in that case we would not want to alter the length of PKs at all

Reply:
We could force users to use `pk_mode: cq-id-only` and `deterministic_cq_id`, but I'm not sure that's the best default.
For example in Azure we hardly ever have more than 1 PK (it's always the ID), same with GCP and `Name` and AWS and `arn`, and so on.

Most tables do have a single field that acts as a PK, and in that case we would use the maximum possible length for the key with this PR. Without this fix we always use `191` which is not the current maximum value.

And it not just about mysql-mysql, it's also Postgres->MySQL and OracleDB->MySql, and maybe Firestore->MySQL.

Reply:
Also, with this fix we won't be shortening the prefix for tables, only making it longer. If the tables exists we would skip it (we don't detect the prefix change) and if it doesn't exist we create it.
It can't exist with a longer prefix, since that's exactly the thing that was preventing it from getting created.

Reply:
Technically we could detect the prefix change, but I believe that would be quite error prone